### ISSUE TITLE
Web Machine Learning Working Group

### ISSUE THREAD

**Issue by dontcallmedom** [w3c/strategy#232]
The discussions at the [Web and Machine Learning workshop](https://www.w3.org/2020/06/machine-learning-workshop/) (see also #150) seems to support the notion that the [Web Neural Network API](https://webmachinelearning.github.io/webnn/) developed by the WebML CG is providing the right kind of primitives to bootstrap wide-scale adoption of machine learning technologies by Web developers.

Early [work on a draft charter](https://github.com/w3c/machine-learning-charter) has thus started - I expect we'll send an advance notice if we get confirmation this is the right direction in the last live session of the workshop.

# Evaluation

* Will this work help to lead the web to its full potential?
Machine learning is reshaping the way a lot of services and applications get built and deployed; making sure the Web provides a privacy-friendly, low latency and efficient platform for these applications seems like an improvement improvement to bring to the platform.

* Is the work Rec-track ready?
  * [Rec Track Readiness](https://www.w3.org/Guide/standards-track/) (AB)
    * Is there a clear problem statement?
Web applications cannot take full advantage of the software and hardware capabilities of the rapidly growing Machine Learning technology stack.

    * Are success criteria explicit?
Bringing interopable Machine Learning primitives to Web browsers that enable usage of hardware acceleration of inference on ML models.

    * Is there a well-socialized proposal to address the problem?
The WebML Community Group has been operating in public for the past two years, with active participation (incl from multiple implementors), with strong presence to W3C TPAC meetings (both as CG and in breakouts). WebNN was at the core of the discussions of the well-attended Web and Machine Learning Workshop over the summer 2020.

    * Has the proposed spec been incubated to reasonable maturity?
The overall orientation of the API as graph-based API that interfaces with OS ML APIs has received early implementation experiences, early TAG attention and seems to match what the ML community sees a useful first step.

    * Is it clear the proposers are not seeking a rubber stamp from W3C?
The discussions at the Web & ML workshop showed a lot of interest in evolving the specification to match the feedback received from the broader community.

    * Are there appropriate expressions of interest?
The active participants in the Community Group show broad interest from the community, incl implementers.

    * Is there actual evidence to back up the answers?
Yes

    * Risks?
ML in the browser brings privacy benefits, but also creates privacy risks since ML can be abused to extract information unbeknownst to the user. ML models are also a known source of biased computing which left unchecked can create significant barriers for many. The ML ecosystem is also evolving fast, so the Working Group will need to ensure it is targeting the right level of primitives to avoid standardizing obsolete or soon-to-be obsolete technologies.

    * Is the timing right?
Given the rapidly growing deployment of ML technologies as a computing paradigm, their broad potential social benefits, it seems urgent for the Web to start building support for this, as many other platforms have already done. Given the current maturity of the work done in the CG and the support it seems to have gathered, this seems like a good time to start the standardization process.

    * Clear RF licensing commitments?
The work has been developed in the Community Group which gives some level of guarantee on RF commitment should this work be moved to the Recommendation Track.

    * Team Engagement?
Yes  

* Do we have the ecosystem of participants needed to make the work successful?
  * users, developers, implementers; industry sectors
Implementors (MS, Google, Apple) are actively involved in the CG.
ML OS APIs (MS, Apple, Google) are actively involved in the CG.
Several ML chips providers (Intel, Apple, Huawei)  are actively involved in the CG
Several ML framework developers (TensorFlow, ONNX, PaddlePaddle, ML5) have been involved in or exposed to the work on the API.

**Comment by wseltzer** [comment]
[Advance notice](https://lists.w3.org/Archives/Public/public-new-work/2020Oct/0001.html)

**Comment by himorin** [comment]
no comment/request from i18n.

**Comment by samuelweiler** [comment]
I've slightly updated the charter template re: separate sections for privacy and security, but no blocking objections to the charter in this form.

FYI, PING conducted an [early privacy review of the spec](https://www.w3.org/Privacy/IG/summaries/PING-minutes-20210121) (minutes not yet posted).  Concerns were expressed about fingerprinting.

**Comment by michael-n-cooper** [comment]
Sorry I overlooked this in my routine review of upcoming charters. I'm responding to a ping from @dontcallmedom. Unfortunately I'm unable to offer substantive input at this time, this is well out of my wheelhouse. The accessibility community has a substantial interest in machine learning, but I'm not able to determine if the proposed spec provokes that interest, or if it's lower level. I would like to run by the APA group, though that may be after W3M has reviewed this.

**Comment by michael-n-cooper** [comment]
APA does want to take a closer look at this, will ask @jasonjgw to review this week.

**Comment by michael-n-cooper** [comment]
Based on what review APA was able to do, there are no comments on this charter. Over to @brewerj to complete accessibility horizontal review.

**Comment by wseltzer** [comment]
Charter sent for AC review: https://lists.w3.org/Archives/Public/public-new-work/2021Feb/0007.html
